hyperspy/signal.py

```diff
@@ -6096,6 +6132,15 @@ def _check_signal_mask(self, mask):
             raise ValueError("The shape of signal mask array must match "
                              "`signal_shape`.")
 
+    def to_gpu(self):
+        if not CUPY_INSTALLED:
+            raise BaseException('cupy is required.')
+        else:
+            self.data = cp.asarray(self.data)
```

Comment:
As mentioned by @din14970 [below](https://github.com/hyperspy/hyperspy/pull/2670#issuecomment-803280165), we should define what to do with lazy signal.
I think that this convenient `to_gpu` method shouldn't be used for lazy signal and it should be changed to raise an error and suggest alternative.